Pre-commit checks, adding simple unit testing & github actions

brucealdridge opened the pull request:
#32 

## Description

Adding the following checks
- JS linting
- Have some of these checks run on pre-commit (similar to WC Payments)
- PHPUnit added and a simple assertTrue(true) unit test running

## How to test this PR

1. `npm install`
2. Change a php file to fail phpcs rules
3. `git commit`

Sample results:
~~~
⇒  git ci -am "#32 Pre-commit checks, adding simple unit testing & github actions"
✔ Preparing...
✔ Running tasks...
✔ Applying modifications...
✔ Cleaning up...
[fix/32-precommit-checks 9068a841] #32 Pre-commit checks, adding simple unit testing & github actions
 21 files changed, 24627 insertions(+), 749 deletions(-)
 create mode 100644 .github/workflows/js-lint-test.yml
 create mode 100644 .husky/.gitignore
 create mode 100755 .husky/pre-commit
 create mode 100755 bin/install-wp-tests.sh
 create mode 100755 bin/phpcs.sh
 create mode 100755 bin/phpunit.sh
 create mode 100755 bin/run-ci-tests.bash
 create mode 100644 lint-staged.config.js
 create mode 100644 phpunit.xml.dist
 create mode 100644 tests/README.md
 create mode 100644 tests/WP_UnitTestCase.php
 create mode 100755 tests/unit/bootstrap.php
 create mode 100644 tests/unit/test-class-simple-pass.php
~~~

## Product impact

- [x] ~Added changelog entry~ (or does not apply)
- [x] Will this PR affect WooCommerce Subscriptions? no
- [x] Will this PR affect WooCommerce Payments? no

brucealdridge committed:
#32 Pre-commit checks, adding simple unit testing & github actions

brucealdridge committed:
Merge branch 'trunk' into fix/32-precommit-checks

brucealdridge committed:
Updating php versions requirements to 7.0+

brucealdridge committed:
set ENV vars for ci

brucealdridge committed:
Remove sarb when running CI to prevent syntax errors on php7.0

brucealdridge committed:
Remove sarb when running CI to prevent syntax errors on php7.0

brucealdridge committed:
Remove sarb when running CI to prevent syntax errors on php7.0

brucealdridge commented:
Currently failing JS listing but this will be resolved by #58

mattallan reviewed (approved):
Awesome changes @brucealdridge ! I'm excited to see this being worked on 😍 

Before merging can you add the following to our `.gitattributes` file so that don't get exported with our tagged versions:

```
.husky export-ignore
tests export-ignore
lint-staged.config.js export-ignore
```

Thanks

Commit:
Include the WP Tests bootstrap which will load core WP functions

While doing a quick tests with some of our helper functions, I noticed the WP core
`current_time()` function wasn't loaded in PHPunit tests.

Instead of manually including the wp-includes/functions.php file, we can use the WP tests bootstrap

Commit:
Add new dev files to gitattributes

brucealdridge commented:
> Awesome changes @brucealdridge ! I'm excited to see this being worked on 😍
> 
> Before merging can you add the following to our `.gitattributes` file so that don't get exported with our tagged versions:
> 
> ```
> .husky export-ignore
> tests export-ignore
> lint-staged.config.js export-ignore
> ```
> 
> Thanks

thanks for doing this!
